### PR TITLE
Standardize chart settings icon buttons

### PR DESCRIPTION
**Pull request by npfitz**
### Description
Updating the icons in some chart settings options to use the `<Button onlyIcon />` component to help standardize hover effects and colors, as well as help them be more semantically correct (they're not clickable buttons rather than clickable icons).

Also took the opportunity to add some unit tests around a related component. Tests don't fully corelate with the change, though it does use one of the changed icons. Existing tests will cover that impacted elements still function the way they should

### How to verify
1. New Question => Orders => Count, Average of Total by Created At
2. Open Viz Settings
3. All icons used (chevron down, ellipsis, close, drag handle) should have consistent hover effects.
4. Questions with multiple dimensions should behave the same way.

### Demo
chrome_BqJLHxUiVy

### CODE REVIEW

File: frontend/test/metabase/visualizations/components/settings/ChartSettingFieldsPicker.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { renderWithProviders, screen } from "__support__/ui";
+
+import ChartSettingFieldsPicker from "metabase/visualizations/components/settings/ChartSettingFieldsPicker";
+
+const DEFAULT_PROPS = {
+  options: [
+    { name: "Count", value: "count" },
+    { name: "Average of Total", value: "avg" },
+  ],
+  colors: { avg: "#A989C5", count: "#509EE3" },
```
Comment (gusaiani) [inline frontend/test/metabase/visualizations/components/settings/ChartSettingFieldsPicker.unit.spec.js:12]: are we really hard-coding the colors like this?
shouldn't we use whitelabelable colors?

File: e2e/test/scenarios/binning/binning-reproductions.cy.spec.js [hunk 2]
```diff
@@ -207,7 +207,9 @@ describe("binning related reproductions", () => {
     cy.findByTestId("sidebar-left").within(() => {
       cy.findByTestId("Table-button").click();
       cy.findByTextEnsureVisible("Table options");
-      cy.findByText("Created At").siblings(".Icon-eye_outline").click();
+      cy.findByText("Created At")
+        .siblings("[data-testid$=hide-button]")
```
Comment (gusaiani) [inline e2e/test/scenarios/binning/binning-reproductions.cy.spec.js:211]: That's quite a peculiar selector with the $ there.
New to me. Can you explain?

Comment (npfitz) [reply e2e/test/scenarios/binning/binning-reproductions.cy.spec.js:211]: It ensures that the `data-testid` ends with that string. The beginning of the testid has the column name in it

File: frontend/src/metabase/visualizations/components/settings/ColumnItem.jsx [hunk 3]
```diff
@@ -11,9 +10,12 @@ import {
   ColumnItemColorPicker,
 } from "./ColumnItem.styled";
 
-const ActionIcon = ({ icon, onClick }) => (
+const ActionIcon = ({ icon, onClick, ...props }) => (
```
Comment (gusaiani) [inline frontend/src/metabase/visualizations/components/settings/ColumnItem.jsx:13]: are we using props for anything?

**Review by gusaiani — APPROVED** [review]
LGTM 👍

File: e2e/test/scenarios/binning/binning-reproductions.cy.spec.js (hunk 2 above)
Comment (gusaiani) [reply e2e/test/scenarios/binning/binning-reproductions.cy.spec.js:211]: ah...makes total sense.
Fancy!

**Pull request merged**